packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -192,21 +192,60 @@ export class SelectedNetworkController extends BaseController<
 
     this.messagingSystem.subscribe(
       'NetworkController:stateChange',
-      (availableNetworkClientIds) => {
-        // if a network is updated or removed, update the networkClientId for all domains
-        // that were using it to the selected network client id
-        const { selectedNetworkClientId } = this.messagingSystem.call(
-          'NetworkController:getState',
-        );
-        Object.entries(this.state.domains).forEach(
-          ([domain, networkClientIdForDomain]) => {
-            if (!availableNetworkClientIds.includes(networkClientIdForDomain)) {
-              this.setNetworkClientIdForDomain(domain, selectedNetworkClientId);
-            }
-          },
+      (
+        { selectedNetworkClientId, networkConfigurationsByChainId },
+        patches,
+      ) => {
+        const patch = patches.find(
+          ({ op, path }) =>
+            (op === 'replace' || op === 'remove') &&
+            path[0] === 'networkConfigurationsByChainId',
```

Comment:
Thoughts on using filter vs find? Not that it does currently, but `patches` could include multiple state changes

Reply:
This whole technique won't work if there are multiple changes.  It relies on the fact that the network controller only exposes an APIs to update or remove a single network at a time.

If multiple networks were updated in 1 update, you wouldn't be able to correlate the missing network client ids to a particular chain id.  Since there's only 1, we can do it with `path[1]`.

Reply:
unless I missed some overload of `subscribe` that gives you enough information to do so